src/main/java/org/web3j/gradle/plugin/Web3jPlugin.java

```diff
@@ -31,7 +31,7 @@ public void apply(final Project target) {
         target.getPluginManager().apply(JavaPlugin.class);
         target.getPluginManager().apply(SolidityPlugin.class);
         target.getExtensions().create(Web3jExtension.NAME, Web3jExtension.class, target);
-        target.getDependencies().add("implementation", "org.web3j:core:3.+");
+        target.getDependencies().add("implementation", "org.web3j:core:4.+");
```

Comment:
If the Gradle plugin and web3j have versions inline with each other, we can use `target.getVersion()` instead of hard-coding the web3j version here.

Reply:
I believe you discussed this with @antonydenyer , is this ok to approve now, @xaviarias ?

Reply:
This was so that when you install the gradle plugin you get the same version of web3j rather than the latest major. For instance, if someone explicitly installed 4.0.1 of the gradle plugin they may get 4.10.1 of web3j.

Reply:
In that case are you happy to approve this PR, whereby the version of the plugin and the web3j library are locked and inline with one-another?

Reply:
@antonydenyer , @xaviarias  this is ready for review. Thanks